code/modules/mob/living/silicon/pai/pai.dm

```diff
@@ -298,8 +298,18 @@
 			pai.radio.attackby(W, user, params)
 		else if(istype(W, /obj/item/encryptionkey))
 			pai.radio.attackby(W, user, params)
-	else
+	else if(istype(W, /obj/item/encryptionkey))
 		to_chat(user, "Encryption Key ports not configured.")
+	if(istype(W, /obj/item/card/id))
+		var/obj/item/card/id/idcard = W
+		if(!pai?.aiPDA.owner) //just in case no name has been set for the PDA
+			pai?.aiPDA.owner = pai.real_name
+			pai?.aiPDA.ownjob = "pAI"
+			pai?.aiPDA.name = pai.real_name + " (pAI)"
+		if(((idcard.registered_name == pai?.real_name) || (idcard.assignment == "pAI")) && idcard.registered_name)
```

Comment:
what even is this check

Reply:
The id card must either have the name field equal to the pAI's name, or the job field equal to pAI, to prevent it being used as a discount agent card/hidden id storage. Additionally, the id card must have a registered name or it will fail to be added to the PDA.